### PR TITLE
add runtime 5.0.10 release notes

### PR DESCRIPTION
**Pull request by paolaperaza**
Based on: https://astronomer.slack.com/archives/C020PVBJPKK/p1666023956020409

### CODE REVIEW

File: astro/runtime-release-notes.md [hunk 1]
```diff
@@ -101,9 +68,25 @@ For a complete list of commits, see the [Apache Airflow 2.4.0 release notes](htt
...
+- Airflow version: 2.3.4
+
+### Additional improvements
+
+- Upgraded `astronomer-providers` to 1.10.0, which includes two new deferrable versions of operators, `SFTPSensorAsync` and `ExternalDeploymentTaskSensorAsync`. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.10.0/CHANGELOG.rst).
+- Upgraded `openlineage-airflow` to version `0.15.1`. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).
+
+### Bug fixes
+
+- Revert “Cache the custom secrets backend so the same instance gets re-used” (#25556)
```
Comment (jwitz) [inline astro/runtime-release-notes.md:86]:
```suggestion
- Revert “Cache the custom secrets backend so the same instance gets re-used” ([#25556](https://github.com/apache/airflow/pull/25556))
```

File: astro/runtime-release-notes.md [hunk 2]
```diff
@@ -101,9 +68,25 @@ For a complete list of commits, see the [Apache Airflow 2.4.0 release notes](htt
...
+- Remove TaskFail duplicates check (#26714)
```
Comment (jwitz) [inline astro/runtime-release-notes.md:87]:
```suggestion
```
This is a check for Runtime only and has no end user impact afaik, I think we can remove it?

File: astro/runtime-release-notes.md [hunk 3]
```diff
@@ -101,9 +68,25 @@ For a complete list of commits, see the [Apache Airflow 2.4.0 release notes](htt
...
+- Fix faulty executor config serialization logic (#26191)
```
Comment (jwitz) [inline astro/runtime-release-notes.md:88]:
```suggestion
```
Same with this, we should either check with a SME on actual user impact and remove the commit # or remove the note entirely.

File: astro/runtime-release-notes.md [hunk 4]
```diff
@@ -286,6 +269,39 @@ Astro Runtime 5.0.0 provides support for [Airflow 2.3.0](https://airflow.apache.
 
 For more information on Airflow 2.3, see ["Apache Airflow 2.3 — Everything You Need to Know"](https://www.astronomer.io/blog/apache-airflow-2-3-everything-you-need-to-know) by Astronomer.
 
+## Astro Runtime 4.2.7
```
Comment (jwitz) [inline astro/runtime-release-notes.md:272]: I'm cool with this for now but we should have a docs discussion about whether we want release notes sorted chronologically or by version #.

**Review by jwitz — APPROVED** [review]
3 minor comments, otherwise LGTM!

File: astro/runtime-release-notes.md (hunk 2 above)
Comment (jedcunningham) [reply astro/runtime-release-notes.md:87]: What do you mean this is a "check for Runtime only"?

This didn't "break" anything afaik, it was just unnecessary and incorrect migration work. Do I have that right @dstandish?

File: astro/runtime-release-notes.md (hunk 3 above)
Comment (jedcunningham) [reply astro/runtime-release-notes.md:88]: This definitely should stay, as it's a fix for KubernetesExecutor executor_config. See the linked issues.

File: astro/runtime-release-notes.md (hunk 2 above)
Comment (jwitz) [reply astro/runtime-release-notes.md:87]: @jedcunningham I'm confused, what's the difference between commits from `airflow-private` and OSS?

Comment (jedcunningham) [reply astro/runtime-release-notes.md]: Generally, nothing. I'm happy to have a chat about it if you'd like the understand the process a bit better. Short version is these are just commits slated for newer Airflow versions that are being applied to older ones.

**Pull request merged**